sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala

```diff
@@ -1050,15 +1052,17 @@ class AnalysisErrorSuite extends AnalysisTest {
     val t2 = LocalRelation(b, c).as("t2")
 
     // SELECT * FROM t1 WHERE a = (SELECT sum(c) FROM t2 WHERE t1.* = t2.b)
-    assertAnalysisError(
+    assertAnalysisErrorClass(
       Filter(EqualTo(a, ScalarSubquery(t2.select(sum(c)).where(star("t1") === b))), t1),
-      "Invalid usage of '*' in Filter" :: Nil
+      expectedErrorClass = "INVALID_USAGE_OF_STAR",
+      expectedMessageParameters = Map("elem" -> "'*'", "prettyName" -> "Filter")
     )
 
     // SELECT * FROM t1 JOIN t2 ON (EXISTS (SELECT 1 FROM t2 WHERE t1.* = b))
-    assertAnalysisError(
+    assertAnalysisErrorClass(
       t1.join(t2, condition = Some(Exists(t2.select(1).where(star("t1") === b)))),
-      "Invalid usage of '*' in Filter" :: Nil
+      expectedErrorClass = "INVALID_USAGE_OF_STAR",
+      expectedMessageParameters = Map("elem" -> "'*'", "prettyName" -> "Filter")
```

Comment:
Could you output a sql statement instead of internal node name, for example `WHERE` instead of `Filter`?

Reply:
Thanks @MaxGekk for review.

I tried to output a SQL statement here, but unfortunately I couldn't proper way to expose the SQL statement rather than node name 😢 

The code path is here:

https://github.com/apache/spark/blob/866343c7be47d71b88ae9a6b4dda26f8c4f5964b/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala#L247-L250

The type of `operator` is `LogicalPlan`, and `s` is `Star` here.

Can we extract the SQL statement from these objects?

Do you happen to help outputting a SQL statement here??

Reply:
I tried to `operator.expressions.map(toPrettySQL).mkString(",")`, but it returns `"(unresolvedstar() = b)"`.

Reply:
@MaxGekk could you happen to take a look at the previous comments when you find some time?
Also cc @cloud-fan for reminder.